### PR TITLE
Add Java solution using Sets to usaco-831

### PR DESCRIPTION
**Pull request by Cubik65536**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [X] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [X] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: solutions/usaco-831.mdx [hunk 1]
```diff
@@ -170,4 +171,67 @@ int main() {
...
+		check(gameboard[0][2], gameboard[1][1], gameboard[2][0]);
+
+		io.println(singleCow.size());
+		io.println(teams.size());
+
+		io.close();
+	}
+}
+
+//CodeSnip{Kattio}
```
Comment (nathangong) [inline solutions/usaco-831.mdx:232]:
```suggestion
	}
	
	// CodeSnip{Kattio}
}
```

File: solutions/usaco-831.mdx [hunk 2]
```diff
@@ -170,4 +171,67 @@ int main() {
...
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class TeamTicTacToe {
+	private static char[][] gameboard = new char[3][3];
+	private static TreeSet<Character> singleCow = new TreeSet<>();
+	private static TreeSet<String> teams = new TreeSet<>();
```
Comment (nathangong) [inline solutions/usaco-831.mdx:184]:
```suggestion
	private static Set<Character> singleCow = new TreeSet<>();
	private static Set<String> teams = new TreeSet<>();
```

File: solutions/usaco-831.mdx [hunk 3]
```diff
@@ -170,4 +171,67 @@ int main() {
...
+
+	private static void check(char first, char second, char third) {
+		// Different Cows in a row/column/diagonal
+        	TreeSet<Character> contained = new TreeSet<>();
```
Comment (nathangong) [inline solutions/usaco-831.mdx:188]:
```suggestion
        	Set<Character> contained = new TreeSet<>();
```

File: solutions/usaco-831.mdx [hunk 4]
```diff
@@ -170,4 +171,67 @@ int main() {
...
+import java.io.*;
+import java.util.*;
+
+public class TeamTicTacToe {
+	private static char[][] gameboard = new char[3][3];
+	private static TreeSet<Character> singleCow = new TreeSet<>();
+	private static TreeSet<String> teams = new TreeSet<>();
+
+	private static void check(char first, char second, char third) {
+		// Different Cows in a row/column/diagonal
```
Comment (nathangong) [inline solutions/usaco-831.mdx:187]:
```suggestion
		// Stores unique cows in a row/column/diagonal
```

File: solutions/usaco-831.mdx [hunk 5]
```diff
@@ -170,4 +171,67 @@ int main() {
...
+        	TreeSet<Character> contained = new TreeSet<>();
+        	contained.add(first);
+        	contained.add(second);
+        	contained.add(third);
+        	if (contained.size() == 1) {
+			// One
```
Comment (nathangong) [inline solutions/usaco-831.mdx:193]:
```suggestion
			// There is only one unique cow
```

File: solutions/usaco-831.mdx [hunk 6]
```diff
@@ -170,4 +171,67 @@ int main() {
...
+		// Different Cows in a row/column/diagonal
+        	TreeSet<Character> contained = new TreeSet<>();
+        	contained.add(first);
+        	contained.add(second);
+        	contained.add(third);
+        	if (contained.size() == 1) {
+			// One 
+			singleCow.add(contained.first());
+		} else if (contained.size() == 2) {
+			// Two
```
Comment (nathangong) [inline solutions/usaco-831.mdx:196]:
```suggestion
			// There are two unique cows
```

**Review by nathangong — CHANGES_REQUESTED** [review]
Nice code! Just some small changes

File: solutions/usaco-831.mdx [hunk 7]
```diff
@@ -170,4 +171,69 @@ int main() {
...
+	private static char[][] gameboard = new char[3][3];
+	private static Set<Character> singleCow = new TreeSet<>();
+	private static Set<String> teams = new TreeSet<>();
+
+	private static void check(char first, char second, char third) {
+		// Stores unique cows in a row/column/diagonal
+        	Set<Character> contained = new TreeSet<>();
+        	contained.add(first);
+        	contained.add(second);
+        	contained.add(third);
```
Comment (nathangong) [inline solutions/usaco-831.mdx:191]: spaces -> tabs, `contained` actually has to be type TreeSet my bad
```suggestion
		TreeSet<Character> contained = new TreeSet<>();
		contained.add(first);
		contained.add(second);
		contained.add(third);
```

Comment (Cubik65536) [reply solutions/usaco-831.mdx:191]: Actually, I'm already using tabs, but it seems that there's some issue when I copy-paste my code from IDE to GitHub, I'll reformat it again.

**Review by nathangong — APPROVED** [review]
Looks good

**Review by dongliu0426 — APPROVED** [review]
thanks!

**Pull request merged**